File: easy/2073_time_needed_to_buy_tickets/solution.py
from collections import deque
from typing import List


class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        queue = deque((i, ticket) for i, ticket in enumerate(tickets))
        time = 0

        while queue:
            i, ticket = queue.popleft()

            time += 1
            ticket -= 1

            if ticket > 0:
                queue.append((i, ticket))
            elif i == k:
                break

        return time

File: easy/2073_time_needed_to_buy_tickets/solution.cpp
#include <vector>

class Solution {
public:
  int timeRequiredToBuy(std::vector<int> &tickets, int k) {
    int time = 0;

    for (size_t i = 0; i < tickets.size(); ++i) {
      time += std::min(tickets[i], tickets[k] - (i > k));
    }

    return time;
  }
};
